tests/scripts/analyze_outcomes.py

```diff
@@ -87,20 +122,66 @@ def read_outcome_file(outcome_file):
                 outcomes[key].failures.append(setup)
     return outcomes
 
-def analyze_outcome_file(outcome_file):
-    """Analyze the given outcome file."""
+def do_analyze_coverage(outcome_file, args):
+    """Perform coverage analyze."""
+    del args # unused
     outcomes = read_outcome_file(outcome_file)
-    return analyze_outcomes(outcomes)
+    results = analyze_outcomes(outcomes)
+    return results.error_count == 0
+
+def do_analyze_driver_vs_reference(outcome_file, args):
+    """Perform driver vs reference analyze."""
+    components = args['components'].split(',')
+    ignored_tests = args['ignored'].split(',')
+    ignored_tests = ['test_suite_' + x for x in ignored_tests]
+    # We need exactly 2 components to analyze (first driver and second reference)
+    if(len(components) != 2 or "accel" not in components[0] or "reference" not in components[1]):
```

Comment:
There's currently no way for users to pass the components, right? So we're sure that they're good?

Reply:
Components for the task are specified in the `TASKS` dictionary.

Reply:
Agreed, I don't think we need that validation code any more. (Especially if we change the dictionary to have one entry per component, as I suggested.)

Reply:
Yes I will clean this up while changing the form of `TASKS`.